src/plugins/intel_gpu/src/graph/slice.cpp

```diff
@@ -26,11 +26,12 @@ inline std::vector<layout> slice_inst::calc_output_layouts(const slice_node&, co
     std::unordered_map<size_t, ov::Tensor> const_data;
     for (std::size_t i = 1; i < impl_param.input_layouts.size(); i++) {
         auto shape_len = shape_size(impl_param.input_layouts[i].get<ShapeType>().to_shape());
-        input_shapes.push_back({static_cast<ov::Dimension::value_type>(shape_len)});
+        const ov::PartialShape input_shape{static_cast<ov::Dimension::value_type>(shape_len)};
+        input_shapes.push_back(input_shape);
```

Comment:
```suggestion
        input_shapes.emplace_back({static_cast<ov::Dimension::value_type>(shape_len)});
```
then in `make_tensor` use `input_shapes.back()`.

It will avoid create additional `PartialShape` which is copied to `input_shapes` vector then copied to `layout` and destroyed at end.

Reply:
It won't work - input_shapes is of type Shape, not PartialShape, and Shape does not have a proper ctor. Even if you could make it, then layout class in make_tensor accepts only PartialShape, not Shape, so you would have additional ctor called.

IMO the way it is right know it is more readable, which is more important than such optimizations, especially  in a code that likely you will never see in any profiling.